File: core/src/main/java/ua/kiev/supersergey/judgement_registry_parser/core/service/KeywordService.java
package ua.kiev.supersergey.judgement_registry_parser.core.service;

import ua.kiev.supersergey.judgement_registry_parser.core.contoller.PaginatedResponse;
import ua.kiev.supersergey.judgement_registry_parser.core.entity.Keyword;

import java.util.List;
import java.util.Optional;

public interface KeywordService {
    PaginatedResponse<List<Keyword>> getAllKeywords(int page, int size);
    Optional<Keyword> findOne(String name);
    Keyword addKeyword(Keyword keyword);
    Keyword updateKeyword(Keyword keyword);
    Keyword deleteKeyword(Keyword k);
    PaginatedResponse<List<Keyword>> findAllByKeyword(String keyword, int page, int size);
}

File: judgement-registry-parser-ui/src/app/keywords/keyword.service.ts
import {Injectable} from '@angular/core';
import {Keyword} from './keyword';
import {Document} from "../document";
import {Observable} from 'rxjs';
import {HttpClient, HttpHeaders} from '@angular/common/http';
import {environment} from '../../environments/environment';
import {MessageService} from "../messages/message.service";
import {Page} from "../page";

const httpOptions = {
  headers: new HttpHeaders({
    'Content-Type': 'application/json',
    'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, OPTIONS',
    'Access-Control-Allow-Origin': '*'
  })
};


@Injectable({
  providedIn: 'root'
})

export class KeywordService {
  requestUrl = `${environment.host}:${environment.port}/${environment.baseUrl}`;
  keyword : Keyword;
  constructor(private http: HttpClient, private messageService : MessageService ) { }


  getKeywords(page: number, size: number) : Observable<Page<Keyword[]>> {
    const endpoint = `/keyword?page=${page}&size=${size}`;
    return this.http.get<Page<Keyword[]>>(this.requestUrl + endpoint);
  }

  getDocumentsByKeyword(name : string): Observable<Document[]> {
    const endpoint = "/keyword/details/" + name;
    return this.http.get<Document[]>(this.requestUrl + endpoint);
  }

  addKeyword(newKeyword: Keyword) : Observable<Keyword> {
    const endpoint = `${this.requestUrl}/keyword/`;
    return this.http.post<Keyword>(endpoint, newKeyword, httpOptions);
  }

  deleteKeyword(keyword: Keyword) : Observable<Keyword> {
    const endpoint = `${this.requestUrl}/keyword/${keyword.keyword}`;
    return this.http.delete<Keyword>(endpoint, httpOptions);
  }

  updateKeyword(keyword: Keyword) {
    const endpoint = `${this.requestUrl}/keyword/`;
    return this.http.put<Keyword>(endpoint, keyword, httpOptions);
  }

  findKeyword(value: string) : Observable<Keyword[]>{
    const endpoint = `${this.requestUrl}/keyword/${value}`;
    return this.http.get<Keyword[]>(endpoint);
  }
}
